src/holofuel/pages/TransactionHistory/TransactionHistory.js

```diff
@@ -17,31 +17,49 @@ import './TransactionHistory.module.css'
 import HashAvatar from '../../../components/HashAvatar/HashAvatar'
 import { OFFER_REQUEST_PATH } from 'holofuel/utils/urls'
 
-// Data - Mutation hooks with refetch:
-
 function usePollCompletedTransactions ({ since }) {
-  const { data: { holofuelNewCompletedTransactions = [] } = {} } = useQuery(HolofuelNewCompletedTransactionsQuery, { fetchPolicy: 'cache-and-network', pollInterval: 5000, variables: { since } })
+  const { data: { holofuelNewCompletedTransactions = [] } = {} } = useQuery(HolofuelNewCompletedTransactionsQuery, { fetchPolicy: 'cache-and-network', pollInterval: 15000, variables: { since } })
   return holofuelNewCompletedTransactions
 }
 
 const FILTER_TYPES = ['all', 'withdrawals', 'deposits', 'pending']
 
 export default function TransactionsHistory ({ history: { push } }) {
-  const { loading: ledgerLoading, data: { holofuelLedger: { balance: holofuelBalance } = {} } = {} } = useQuery(HolofuelLedgerQuery, { fetchPolicy: 'cache-and-network', pollInterval: 5000 })
-  const { loading: loadingPendingTransactions, data: { holofuelWaitingTransactions = [] } = {} } = useQuery(HolofuelWaitingTransactionsQuery, { fetchPolicy: 'cache-and-network' })
+  const { loading: ledgerLoading, data: { holofuelLedger: { balance: holofuelBalance } = {} } = {} } = useQuery(HolofuelLedgerQuery, { fetchPolicy: 'cache-and-network', pollInterval: 15000 })
+  const { loading: loadingPendingTransactions, data: { holofuelWaitingTransactions = [] } = {} } = useQuery(HolofuelWaitingTransactionsQuery, { fetchPolicy: 'cache-and-network', pollInterval: 15000 })
   const { loading: loadingCompletedTransactions, data: { holofuelCompletedTransactions = [] } = {} } = useQuery(HolofuelCompletedTransactionsQuery, { fetchPolicy: 'cache-and-network' })
 
   const since = !isEmpty(holofuelCompletedTransactions) ? holofuelCompletedTransactions[0].timestamp : ''
   const pollingResult = usePollCompletedTransactions({ since })
 
-  const completedTransactions = holofuelCompletedTransactions.concat(pollingResult)
-  const filteredTransactionById = intersectionBy('id', completedTransactions, holofuelWaitingTransactions)
-  const pendingTransactions = reject(({ id }) => find({ id }, filteredTransactionById), holofuelWaitingTransactions)
+  const filteredTransactionById = intersectionBy('id', holofuelCompletedTransactions, pollingResult)
+  const filteredPollingResult = reject(({ id }) => find({ id }, filteredTransactionById), pollingResult)
+  const completedTransactions = holofuelCompletedTransactions.concat(filteredPollingResult)
+  const pendingTransactions = holofuelWaitingTransactions.filter(pendingTx => pendingTx.status !== STATUS.completed)
 
   const goToCreateTransaction = () => push(OFFER_REQUEST_PATH)
 
   const [filter, setFilter] = useState(FILTER_TYPES[0])
 
+  function useLoadingFirstTime (loading) {
```

Comment:
let's move this into utils now that we're using it in two files (and could also use it in Home replacing what's on line 37 in this PR)